src/components/frame/Extensions/pages/ExtensionDetailsPage/utils.tsx

```diff
@@ -209,3 +294,74 @@ export const mapExtensionActionPayload = (
     {},
   );
 };
+
+export const getTextChunks = () => {
+  const HeadingChunks = (chunks: React.ReactNode[]) => (
+    <h4 className="mb-4 mt-6 font-semibold text-gray-900">{chunks}</h4>
+  );
+
+  const ParagraphChunks = (chunks: React.ReactNode[]) => (
+    <p className="mb-4">{chunks}</p>
+  );
+  const BoldChunks = (chunks: React.ReactNode[]) => (
+    <b className="font-medium text-gray-900">{chunks}</b>
+  );
+  const ListChunks = (chunks: React.ReactNode[]) => (
+    <ul className="my-4 ml-4 list-disc">{chunks}</ul>
+  );
+  const ListItemChunks = (chunks: React.ReactNode[]) => <li>{chunks}</li>;
+
+  return {
+    h4: HeadingChunks,
+    p: ParagraphChunks,
+    b: BoldChunks,
+    ul: ListChunks,
+    li: ListItemChunks,
+  };
+};
+
+export const getExtensionParams = (extensionData: AnyExtensionData | null) => {
+  if (!extensionData) {
+    return {};
+  }
+
+  const isExtensionInstalled = isInstalledExtensionData(extensionData);
+  const { initializationParams = [] } = extensionData;
+  const initialValues = createExtensionSetupInitialValues(initializationParams);
+
+  if (isExtensionInstalled) {
+    switch (extensionData.extensionId) {
+      case Extension.StakedExpenditure: {
+        return extensionData.params?.stakedExpenditure?.stakeFraction
+          ? {
+              stakeFraction: getStakeFraction(
+                extensionData.params.stakedExpenditure.stakeFraction,
+              ),
+            }
+          : initialValues;
+      }
+      default: {
+        return extensionData.params?.[camelCase(extensionData.extensionId)];
+      }
+    }
+  }
+
+  return initialValues;
+};
+
+export const getActionData = (extensionData: AnyExtensionData) => {
+  const isExtensionInstalled = isInstalledExtensionData(extensionData);
+
+  switch (extensionData.extensionId) {
+    case Extension.StakedExpenditure: {
+      if (isExtensionInstalled && extensionData.isEnabled) {
+        return ActionTypes.SET_STAKE_FRACTION;
+      }
```

Comment:
I might be wrong but I think this is where the confusion around #2498 could have aroused from. Both contracts and the saga should support deprecating but it would be prevented by this if check since deprecated extension has its `isEnabled` set to false.
@arrenv

Reply:
@jakubcolony I don't understand. There's a separate saga that is used to reenable the extension. Please check `src/components/frame/Extensions/pages/ExtensionDetailsPage/partials/ExtensionDetails/hooks.tsx`. 
![image](https://github.com/JoinColony/colonyCDapp/assets/44369011/a501db31-c7e3-4ece-9138-622099ebd8d4)

Reply:
@joanna-pagepro I'm referring to the issue #2498 which states:
> It should be possible to edit the "Required stake" percentage value when the extension is in a deprecated state. This is possible on the contracts, however, according to recent updates as of this PR https://github.com/JoinColony/colonyCDapp/pull/2474, it is currently not supported by the saga.

I don't think this is correct, the `setStakeFraction` saga supports editing stake fraction for deprecated extensions. It is the UI that needs to be modified to allow that.

The if check I commented on would be one of the things that might need to change, as currently it either sets the stake fraction (for non-deprecated extensions) or enables the deprecated ones. There is no path to set stake fraction for deprecated extension without enabling it.

Nonetheless, I just wanted to mention if for future reference, it's out of scope of this PR.

Reply:
@jakubcolony I guess that we have a small misunderstanding here. I asked @arrenv if a user should be able to change the stake fraction while reenabling the extension and not if he should be able to change the stake fraction while the extension is deprecated without reenabling it. As I understand from your comment, you want a user to be able to modify the stake fraction while in the deprecated state without reenabling it.

Reply:
@joanna-pagepro thanks for explaining, I have checked with Arren and being able to set stake fraction for a deprecated extension is one of the requirements and should be part of this PR. 

> I asked @arrenv if a user should be able to change the stake fraction while reenabling the extension and not if he should be able to change the stake fraction while the extension is deprecated without reenabling it. As I understand from your comment, you want a user to be able to modify the stake fraction while in the deprecated state without reenabling it.

I think the exact behaviour needs to be confirmed with Arren, the point I was trying to make is that it's ready to be implemented now, and doesn't require any saga modifications.